### PR TITLE
feat: Add WDA with W3C protocol support

### PR DESCRIPTION
**Pull request by mykola-mokhnach**

### CODE REVIEW

File: lib/driver.js [hunk 1]
```diff
@@ -910,31 +910,32 @@ class XCUITestDriver extends BaseDriver {
...
-      desired.desiredCapabilities.defaultAlertAction = 'accept';
+      wdaCaps.defaultAlertAction = 'accept';
     } else if (this.opts.autoDismissAlerts) {
-      desired.desiredCapabilities.defaultAlertAction = 'dismiss';
+      wdaCaps.defaultAlertAction = 'dismiss';
     }
 
-    await this.proxyCommand('/session', 'POST', desired);
+    await this.proxyCommand('/session', 'POST', {
+      firstMatch: [wdaCaps],
```
Comment (KazuCocoa) [inline lib/driver.js]: should be `{capabilities: { firstMatch: [wdaCaps], alwaysMatch: {}, }}`?

Comment (mykola-mokhnach) [reply lib/driver.js]: it will be after https://github.com/appium/WebDriverAgent/pull/203 is published and picked up

File: lib/driver.js [hunk 2]
```diff
@@ -910,31 +910,34 @@ class XCUITestDriver extends BaseDriver {
...
-      desired.desiredCapabilities.defaultAlertAction = 'accept';
+      wdaCaps.defaultAlertAction = 'accept';
     } else if (this.opts.autoDismissAlerts) {
-      desired.desiredCapabilities.defaultAlertAction = 'dismiss';
+      wdaCaps.defaultAlertAction = 'dismiss';
     }
 
-    await this.proxyCommand('/session', 'POST', desired);
+    await this.proxyCommand('/session', 'POST', {
+      capabilities: {
```
Comment (KazuCocoa) [inline lib/driver.js]: 👍

File: lib/wda/webdriveragent.js [hunk 3]
```diff
@@ -281,7 +281,6 @@ class WebDriverAgent {
     this.proxyReqRes = this.jwproxy.proxyReqRes.bind(this.jwproxy);
 
     this.noSessionProxy = new NoSessionProxy(proxyOpts);
-    this.noSessionProxyReqRes = this.noSessionProxy.proxyReqRes.bind(this.noSessionProxy);
```
Comment (KazuCocoa) [inline lib/wda/webdriveragent.js]: 👍

**Pull request merged**